### DISCUSSION TITLE
Make package.json optional in a mono-repo setup

### DISCUSSION THREAD

**Discussion by squiddy** [jaredpalmer/razzle#1611] [Ideas & RFCs]
The `with-monorepo` example is similar to what we're doing for our project, however both "razzle apps" share the same set of dependencies that are setup in the `package.json` of the repository root.

Right now, I have to put at least an empty `package.json` into each app's folder. Some testing with the example suggests that there might be two places that require a package.json to be present.

I'm wondering if our case is special, or if making that file optional is a worthy goal? Or perhaps, alternatively, have it fall back to the parent package.json?

**Comment by fivethreeo** [comment]
If you just don't add those dirs to workspaces in the root package.json I don't think those package.jsons are needed. But the idea was to add the production deps to the workspace dirs so they can be deployed separately.

Also not sure how resolving works then if you don't deploy the root aswell.

**Reply by squiddy** [reply]
> But the idea was to add the production deps to the workspace dirs so they can be deployed separately.

I see. Our setup is a bit differently then. Our apps share the same dependencies (same package.json), but are deployed separately. We're not doing anything in `workspaces` right now. We build one npm package that includes the `build` output of both applications.

Running it then is a matter of `node app1/build/server.js` and `node app2/build/server.js`.

---

I was asking because there seem to be only two places that actually require a package.json to be present when we make use of `RAZZLE_APP_PATH`. 

https://github.com/jaredpalmer/razzle/blob/92ef4c2667e2f8123085ca3893d42dd18b9a96a5/packages/razzle/config/paths.js#L26
https://github.com/jaredpalmer/razzle/blob/92ef4c2667e2f8123085ca3893d42dd18b9a96a5/packages/razzle/config/createJestConfig.js#L69

---

Not a big issue though, I understand the intention better now. We can probably close this. :)

**Comment by fivethreeo** [comment]
I see package.json is required some places. We could test for them and supply default values if there is no json.

**Reply by squiddy** [reply]
Happy to provide a PR later today.
